doc/did-method-spec.md

```diff
@@ -0,0 +1,286 @@
+did:ethr method
+=================
+
+## Author
+
+- uPort Team: <https://www.uport.me/contact>
+
+## Preface
+
+The ethr DID method specification conforms to the requirements specified in 
+the [DID specification](https://w3c-ccg.github.io/did-spec/), currently published by the 
+W3C Credentials Community Group. For more information about DIDs and DID method specifications, 
+please see the [DID Primer](https://github.com/WebOfTrustInfo/rebooting-the-web-of-trust-fall2017/blob/master/topics-and-advance-readings/did-primer.md)
+
+## Abstract
+Decentralized Identifiers (DIDs, see [1]) are designed to be compatible with any distributed ledger or network.
+In the Ethereum community, a pattern is known as ERC1056 (see [2]) utilizes a smart contract for a lightweight
+identity management system intended explicitly for off-chain usage.
+
+We propose a new DID method that allows any Ethereum account regardless if a smart contract or key pair,
+to become a valid identity. An identity needs no registration. In the case of key management or additional
+attributes such as "service endpoints" are required, we deployed ERC1056 smart contracts on:
+
+* Mainnet: `0xdca7ef03e98e0dc2b855be647c39abe984fcf21b`
+* Ropsten: `0xdca7ef03e98e0dc2b855be647c39abe984fcf21b`
+* Rinkeby: `0xdca7ef03e98e0dc2b855be647c39abe984fcf21b`
+* Kovan: `0xdca7ef03e98e0dc2b855be647c39abe984fcf21b`
+
+Since each Ethereum transaction must be funded, there is a growing trend of on-chain transactions that are
+authenticated via an externally created signature and not by the actual transaction originator. This allows for
+3rd party funding services, or for receivers to pay without any fundamental changes to the underlying
+Ethereum architecture. These kinds of transactions have to be signed by an actual key pair and thus cannot be used 
+to represent smart contract based Ethereum accounts. ERC1056 proposes a way of a smart contract or regular key pair 
+delegating signing for various purposes to externally managed key pairs. This allows a smart contract to be
+represented, both on-chain as well as off-chain or in payment channels through temporary or permanent delegates.
+
+For a reference implementation of this DID method specification see [3].
+
+### Identity Ownership
+Each identity has a single address which maintains ultimate control over it. By default, each identity is controlled
+by itself. As ongoing technological and security improvements occur, an owner can replace themselves with any other
+Ethereum address, such as an advanced multi-signature contract.
+
+There is only ever a single identity owner. More advanced ownership models are managed through a multi-signature
+contract.
+
+## Target System
+
+The target system is the Ethereum network where the ERC1056 is deployed. This could either be:
+- Mainnet
+- Ropsten
+- Rinkeby
+- Kovan
+- other EVM-compliant blockchains such as private chains, or consortium chains.
```

Comment:
Since there are multiple possible target systems, maybe include a brief description of the parameters needed to specify which one is used by an identity. Is that just network id, or would that require an ethereum rpc url etc as well?

Reply:
The problem is that the ethr DID does not have the means to support multiple target systems because we don't encode the system in the DID. For political reasons, i.e., providing a statement that we support any of these chains, I intentionally tried to leave this as open as possible. Our MNID is not part of the DID yet. Our RPC URL only relates to Ethereum transaction signing.

Reply:
@mi-xu what do you think?

Reply:
Ok that makes sense to me.  I think having it default to main net and adding the network as an optional parameter is a sensible solution.